README.md

```diff
@@ -1,54 +1,181 @@
-# ACME webhook example
+# NS1 Webhook for Cert Manager
 
-The ACME issuer type supports an optional 'webhook' solver, which can be used
-to implement custom DNS01 challenge solving logic.
+This is a webhook solver for [NS1](http://ns1.com).
 
-This is useful if you need to use cert-manager with a DNS provider that is not
-officially supported in cert-manager core.
+## Prerequisites
 
-## Why not in core?
+[certmanager](https://github.com/jetstack/cert-manager)
 
-As the project & adoption has grown, there has been an influx of DNS provider
-pull requests to our core codebase. As this number has grown, the test matrix
-has become un-maintainable and so, it's not possible for us to certify that
-providers work to a sufficient level.
+tested with v0.13.0
 
-By creating this 'interface' between cert-manager and DNS providers, we allow
-users to quickly iterate and test out new integrations, and then packaging
-those up themselves as 'extensions' to cert-manager.
+## Installation
 
-We can also then provide a standardised 'testing framework', or set of
-conformance tests, which allow us to validate the a DNS provider works as
-expected.
+Install with helm:
 
-## Creating your own webhook
+```bash
+$ helm install --namespace cert-manager --name cert-manager-webhook-ns1 \
+  ./deploy/cert-manager-webhook-ns1
+```
+
+Issuer
+
+1. Populate a secret with your NS1 API Key:
+```bash
+$ kubectl --namespace cert-manager create secret generic \
+  ns1-credentials --from-literal=apiKey='Your NS1 API Key'
+```
 
-Webhook's themselves are deployed as Kubernetes API services, in order to allow
-administrators to restrict access to webhooks with Kubernetes RBAC.
+2. Add the following resources, something like:
+```bash
+  kubectl --namespace cert-manager apply -f my_resource.yaml
+```
 
-This is important, as otherwise it'd be possible for anyone with access to your
-webhook to complete ACME challenge validations and obtain certificates.
+Note that it may make more sense in your setup to use e.g. `ClusterRole` or
+`ClusterIssuer` and/or to have more nuanced namespace management.
+
+3. Grants permission for service-account to get the secret:
```

Comment:
on the fence. feels reasonable to do something like that, but I also want to keep changes from the "base" example repo really clear and this is the way they did it. It would be nice to template them a bit, but also having them here highlights that there's some flexibility. Don't think it's a huge deal either way, but i'm still thinkng and waffling about it.

Reply:
yeah fair, maybe we provide a fully functional bash cmd then with some `echo << EOF . . . yaml stuff . . . ` stuff or whatever that syntax is

Reply:
maybe, but that's a bit weird cause they need at least a bit of editing. In any event, I'm definitely going to try to make this a little nicer on the next pass when i'm sprucing up the joint some more.

Reply:
I left the yaml clauses in readme, with some reorganization. There's some flexibilty/choice involved with them, so I felt it was better not to moosh 'em into the chart